src/Money.php

```diff
@@ -357,6 +357,14 @@ public function allocate(array $ratios)
         $total = array_sum($ratios);
 
         foreach ($ratios as $ratio) {
+            if ($ratio < 0) {
```

Comment:
@frederikbosch Any of ratios can be `0`, but not all.

1. All ratios must be non-negative (>= 0);
2. Sum of ratios must be greater than `0` (division by zero occur if sum is 0).